src/core/CMakeLists.txt

```diff
@@ -1000,6 +1009,8 @@ INCLUDE_DIRECTORIES(
   metadata
   pal
   processing
+  providers
```

Comment:
No, the provider is not exposed through the bindings, so end use remains unchanged. What's changed here is the requirement that all providers are loaded via libraries. Now it's possible to register providers directly with function pointers. So this should be moderatly easy to expand and expose to the bindings, hence allowing pure python data providers.